src/lib.rs

```diff
@@ -82,28 +82,39 @@ pub mod write;
 pub struct Compression(u32);
 
 impl Compression {
-    /// Create a new compression spec with a specific numeric level (0-9).
+    /// Create a new compression spec with a specific numeric level in the range `1..=9`.
+    #[deprecated(since = "0.5.1", note = "use try_new instead")]
     pub fn new(level: u32) -> Compression {
         Compression(level)
```

Comment:
I think panicking if the compression level is not in the range 1..=9 is fine. If you try to use the returned `Compression` value when compressing data you will get an error anyway.

Reply:
right you get an error, not a panic. 

Anyhow, existing users will already not run into this (or they would have before), and now new users should never run into this behavior, and then at some point we can just drop it from the public interface.

Reply:
I think `Compression::new(2)` is a much better API than `Compression::try_new(2).unwrap()`, so I did much rather fix `Compression::new()` than add a new `Compression::try_new()` and remove the former in the future.

> Anyhow, existing users will already not run into this (or they would have before)

That is exactly why I think adding a panic in `Compression::new()` is fine.